src/components/grouped-character/grouped-character-test.stories.tsx

```diff
@@ -77,3 +78,30 @@ NewValidation.args = {
   separator: "-",
   ...getCommonTextboxArgs(),
 };
+
+export const GroupedCharacterComponent = ({
+  onChange,
+  ...props
+}: {
+  onChange?: (modifiedEvent: any) => void;
+}) => {
+  const [state, setState] = React.useState("");
+
+  const setValue = ({ target }: CustomEvent) => {
+    setState(target.value.rawValue);
+    if (onChange) {
+      onChange(target);
+    }
+  };
+
+  return (
+    <GroupedCharacter
+      label="GroupedCharacter"
+      value={state}
+      onChange={setValue}
+      groups={[2, 2, 3]}
+      separator="-"
+      {...props}
```

Comment:
If you modify the typing of this mock component's props using my suggestion, the typescript compiler will complain that its possible to overwrite the `groups` and `separator` props here, since these could be present in the `props` object.

To fix this, we'll need to pass the `groups` and `separator` props to `GroupCharacter` if they have been defined, else set a default value for them, something like so:

```tsx
export const GroupedCharacterComponent = ({
  onChange,
  groups,
  separator,
  ...props // this will no longer include `groups` and `separator`, since we have already unpacked them in the above two lines
}: GroupedCharacterProps) => {

  /** ...other code... */

  return (
    <GroupedCharacter
      label="GroupedCharacter"
      value={state}
      onChange={setValue}
      groups={groups || [2, 2, 3]} // pass in `groups` if its defined, else set to a default value
      separator={separator || "-"} // pass in `separator` if its defined, else set to a default value
      {...props}
    />
  );
};
```